app/controllers/cloud_subnet_controller.rb

```diff
@@ -12,4 +12,253 @@ class CloudSubnetController < ApplicationController
+        add_flash(_("%{model} no longer exists.") % {:model => ui_lookup(:table => "cloud_subnet")}, :error)
+      else
+        valid_delete, delete_details = subnet.validate_delete_subnet
+        if valid_delete
+          subnets_to_delete.push(subnet)
+        else
+          add_flash(_("Couldn't initiate deletion of %{model} \"%{name}\": %{details}") % {
+            :model   => ui_lookup(:table => 'cloud_subnet'),
+            :name    => subnet.name,
+            :details => delete_details}, :error)
+        end
+      end
+    end
+    unless subnets_to_delete.empty?
+      process_cloud_subnets(subnets_to_delete, "destroy")
+      # Todo: To replace with targeted refresh when avail. or either use tasks
+      EmsRefresh.queue_refresh(ManageIQ::Providers::NetworkManager)
+    end
+
+    # refresh the list if applicable
+    if @lastaction == "show_list"
+      show_list
+      @refresh_partial = "layouts/gtl"
+    elsif @lastaction == "show" && @layout == "cloud_subnet"
+      @single_delete = true unless flash_errors?
+      if @flash_array.nil?
+        add_flash(_("The selected %{model} was deleted") % {:model => ui_lookup(:table => "cloud_subnet")})
+      end
+    end
+  end
+
+  def edit
+    assert_privileges("cloud_subnet_edit")
+    @subnet = find_by_id_filtered(CloudSubnet, params[:id])
+    @in_a_form = true
+    drop_breadcrumb(
+      :name => _("Edit %{model} \"%{name}\"") % {:model => ui_lookup(:table => 'cloud_subnet'), :name => @subnet.name},
+      :url  => "/cloud_subnet/edit/#{@subnet.id}"
+    )
+  end
+
+  def get_checked_subnet_id(params)
+    if params[:id]
+      checked_subnet_id = params[:id]
+    else
+      checked_subnets = find_checked_items
+      checked_subnet_id = checked_subnets[0] if checked_subnets.length == 1
+    end
+    checked_subnet_id
+  end
+
+  def update
+    assert_privileges("cloud_subnet_edit")
+    @subnet = find_by_id_filtered(CloudSubnet, params[:id])
+
+    case params[:button]
+    when "cancel"
+      cancel_action(_("Edit of %{model} \"%{name}\" was cancelled by the user") % {
+        :model => ui_lookup(:table => 'cloud_subnet'),
+        :name  => @subnet.name
+      })
+
+    when "save"
+      form_params
+      begin
+        @subnet.save
+        add_flash(_("Updating %{model} \"%{name}\"") % {
+          :model => ui_lookup(:table => 'cloud_subnet'),
+          :name  => @subnet.name
+        })
+      rescue => e
+        add_flash(_("Unable to update %{model} \"%{name}\": %{details}") % {
+          :model   => ui_lookup(:table => 'cloud_subnet'),
+          :name    => @subnet.name,
+          :details => e
+        }, :error)
+      end
+
+      @breadcrumbs.pop if @breadcrumbs
+      session[:edit] = nil
+      session[:flash_msgs] = @flash_array.dup if @flash_array
+      javascript_redirect :action => "show", :id => @subnet.id
+    end
+  end
+
+  private
+
+  def form_params
+    params[:ip_version] ||= "4"
+    params[:dhcp_enabled] ||= "True"
```

Comment:
`"True"` looks like a typo.. 

Also seems like `dhcp_enabled` is never actually in params, no corresponding field anywhere..